returnn/tf/layers/rec.py

```diff
@@ -524,10 +527,13 @@ def _get_input(self):
     """
     assert self.input_data
     if self.input_data.have_time_axis():
-      x = self.input_data.placeholder  # (batch,time,dim) or (time,batch,dim)
+      x = self.input_data.placeholder  # (batch,time,dim) or (time,batch,dim) or (batch,dim,time)
```

Comment:
Why do you make it so complicated? Just introduce a function `copy_as_time_batch_major` or so. `get_placeholder_as_time_major` currently assumes (just as this comment here suggests) that it is (B,T,D) or (T,B,D), which is incorrect. But if you just have a function like `copy_as_time_batch_major`, it would always be correct. You should not check all the axes in here and make the code so complicated here. Such a function `copy_as_time_batch_major` should do that logic.

Reply:
I did not make anything complicated here, I just extended the assert here. Yes, the whole code could be removed completely if there is a `copy_as_time_batch_major`.